test/unit/math/mix/meta/is_dense_dynamic_test.cpp

```diff
@@ -0,0 +1,67 @@
+#include <stan/math/mix.hpp>
+#include <test/unit/math/mix/meta/eigen_utils.hpp>
+#include <Eigen/Sparse>
+#include <gtest/gtest.h>
+
+TEST(MathMetaPrim, is_dense_dynamic_hierarchy_tests) {
+  using Eigen::Array;
+  using Eigen::Matrix;
+  using stan::is_dense_dynamic;
+  using stan::math::test::all_eigen_dense;
+  all_eigen_dense<false, true, true, true, true, is_dense_dynamic, Matrix, -1,
```

Comment:
Why is it that this isn't true for `EigenBase` types? (the first template argument is false)

Reply:
`EigenBase` does not have a `RowsAtCompileTime` or `ColsAtCompileTime`